### PR TITLE
Refactoring jest-playwright

### PR DESCRIPTION
**Pull request by mmarkelov**
Searching nice and clear solution to run tests for multiple browsers. Trying to do it with custom runner, that will run tests for all browsers. So example for this test:
```js
beforeAll(async () => {
    await page.goto('https://google.com/')
})

it('Google',  async(done) => {
    const title = await page.title()
    expect(title).toBe('Google')
    done()
})
```

will run 3 times: for **chromium**, **webkit** and **firefox**. I can create new folder with small repro of this configuration and some tests in it.

### CODE REVIEW

**Comment by mmarkelov** [comment]
Just made [repo](https://github.com/mmarkelov/refactored-jest-playwright)
You can check it out

**Comment by mmarkelov** [comment]
So it will be breaking change. I decided to use custom runner for testing. I suppose it should be in separate package. 
I just replace whole multiple browser testing from buggy solution to more clear solution with extended **jest-runner**, also I decided to use **jest-circus** as default test-runner.
It will be just **browsers** and **devices** options in configuration.
I suppose that it can be some problems with it, but it is more clean and nice solution for me =)

File: src/PlaywrightRunner.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+const getTests = (
+  browsers: BrowserType[],
+  devices: string[],
+  tests: Test[],
+): Test[] => {
+  let browserTests: Test[] = []
+  browsers.forEach((browser) => {
+    checkBrowserEnv(browser)
+    devices.length
```
Comment (mxschmitt) [inline src/PlaywrightRunner.ts:55]: not: not totally readable in my mind.

**Comment by mxschmitt** [comment]
Maybe we can make a prerelease out of it? So we can try to adopt it in our projects and then we can make an official normal release.

**Comment by mmarkelov** [comment]
> great! I like it :) do we have a lot of breaking changes? the kind of describe / it is probably the same right? Just browser->browsers device->devices

Youp! I suppose so.
> Maybe we can make a prerelease out of it? So we can try to adopt it in our projects and then we can make an official normal release.

 Make a prerelease is a good idea

**Pull request merged**